aws/resource_aws_s3_bucket_test.go

```diff
@@ -2912,9 +2893,7 @@ func testAccCheckAWSS3BucketReplicationRules(n string, providerF func() *schema.
 			}
 		}
 
-		provider := providerF()
-
-		conn := provider.Meta().(*AWSClient).s3conn
+		conn := testAccProvider.Meta().(*AWSClient).s3conn
```

Comment:
We previously did and always should reference the main provider instance in this type of testing for simplicity.